src/Evolution/Systems/Burgers/Subcell/TimeDerivative.hpp

```diff
@@ -0,0 +1,213 @@
+#include "NumericalAlgorithms/Spectral/Mesh.hpp"
+#include "Utilities/ErrorHandling/Assert.hpp"
+#include "Utilities/FakeVirtual.hpp"
+#include "Utilities/Gsl.hpp"
+#include "Utilities/MakeArray.hpp"
+
+namespace Burgers::subcell {
+/*!
+ * \brief Compute the time derivative on the subcell grid using FD
+ * reconstruction.
+ *
+ * The code makes the following unchecked assumptions:
+ * - Assumes Cartesian coordinates with a diagonal Jacobian matrix
+ * from the logical to the inertial frame
+ * - Assumes the mesh is not moving (grid and inertial frame are the same)
+ */
+struct TimeDerivative {
+  template <typename DbTagsList>
+  static void apply(
+      const gsl::not_null<db::DataBox<DbTagsList>*> box,
+      const InverseJacobian<DataVector, 1, Frame::ElementLogical, Frame::Grid>&
+          cell_centered_logical_to_grid_inv_jacobian,
+      const Scalar<DataVector>& /*cell_centered_det_inv_jacobian*/) {
+    // subcell is currently not supported for external boundary elements
+    const Element<1>& element = db::get<domain::Tags::Element<1>>(*box);
+    ASSERT(element.external_boundaries().size() == 0,
+           "Can't have external boundaries right now with subcell. ElementID "
+               << element.id());
+
+    using evolved_vars_tags = typename System::variables_tag::tags_list;
+    using fluxes_tags = typename Fluxes::return_tags;
+
+    // The copy of Mesh is intentional to avoid a GCC-7 internal compiler error.
+    const Mesh<1> subcell_mesh =
+        db::get<evolution::dg::subcell::Tags::Mesh<1>>(*box);
+    const size_t num_reconstructed_pts =
+        subcell_mesh.number_of_grid_points() + 1;
+
+    const Burgers::fd::Reconstructor& recons =
+        db::get<Burgers::fd::Tags::Reconstructor>(*box);
+
+    const auto& boundary_correction =
+        db::get<evolution::Tags::BoundaryCorrection<System>>(*box);
+    using derived_boundary_corrections =
+        typename std::decay_t<decltype(boundary_correction)>::creatable_classes;
+
+    // Variables to store the boundary correction terms on FD subinterfaces
+    std::array<Variables<evolved_vars_tags>, 1> fd_boundary_corrections{};
+
+    // package the data and compute the boundary correction
+    tmpl::for_each<derived_boundary_corrections>(
+        [&boundary_correction, &box, &element, &fd_boundary_corrections,
+         &num_reconstructed_pts, &recons,
+         &subcell_mesh](auto derived_correction_v) {
+          using derived_correction =
+              tmpl::type_from<decltype(derived_correction_v)>;
+          if (typeid(boundary_correction) == typeid(derived_correction)) {
+            using dg_package_field_tags =
+                typename derived_correction::dg_package_field_tags;
+            using dg_package_data_argument_tags =
+                tmpl::append<evolved_vars_tags, fluxes_tags>;
+
+            // Variables that need to be reconstructed for dg_package_data()
+            auto package_data_argvars_lower_face =
```

Comment:
No special reason. Just thought the name might be clearer when I was writing code..?

Reply:
No special reason. Just thought those names might be clearer when I was writing code..? One reason was that GRMHD has `vars_upper_face` and `vars_on_lower_face` so I thought it was better to make a more distinction between names

Reply:
okay. Do you think it would make sense to change the GRMHD names in the same PR to make it consistent? Hopefully then the clearer naming will get used more in new systems?

Reply:
Sure :)